### ISSUE TITLE
Use predictable (semantic?) versioning scheme

### ISSUE THREAD

**Issue by wdconinc** [JeffersonLab/JANA2#139]
Jana2 v2.0.6 is not merely a bugfix release of v2.0.5. Downstream EICrecon code relies on new functionality (e.g. `JANA/JEventProcessorSequentialRoot.h`) introduced in v2.0.6. Could we clarify the versioning scheme here, so it can be used for predictable requirements in downstream projects such as EICrecon? I would suggest semantic versioning here, which would indicate a new minor release for added functionality, and adding `find_package(JANA 2.1 REQUIRED)` to EICrecon.

**Comment by nathanwbrei** [comment]
We are loosely SemVer. In the best of cases, SemVer is approximate. [1] [2] We'll be more disciplined about adhering to SemVer principles as we settle into a more comfortable development pace. The overhead isn't worth it between now and October.
 
CMake thinks it can be a package manager as well as a build system a la Maven, but it is deluding itself. If you pin your dependency versions at the CMake level, you are just creating more headaches for everyone. Don't do that. Pin your dependency versions at the git level. 

[1] https://hynek.me/articles/semver-will-not-save-you/
[2] https://snarky.ca/why-i-dont-like-semver/

**Comment by nathanwbrei** [comment]
@faustus123 On further reflection, we aren't _quite_ SemVer and we will have to address this at some point. Currently, _at the CMake level_, the project is named "JANA" and the major version is 2. The problem is that this implies that any breaking change to the API turns this into JANA3 (at least from the perspective of CMake). A more accurate schema would have the project name be "JANA2" and the major version be 1. (This would communicate that this is a clean rewrite of JANA1 and while we don't plan on doing any more clean rewrites, we will obviously be making breaking changes at some point.) On the other hand, at the end of the day it doesn't matter if our major versions start at 1 or at 2, as long as we increment them every time we make breaking changes. 

Luckily, we named the git repo JANA2, so we don't have this problem at the git level. 

In summary: We need to rename the CMake project name from JANA to JANA2, and this is going to be annoying.

**Comment by faustus123** [comment]
Version matching is supposed to be handled by the JANAConfigVersion.cmake file and the user (e.g. EICrecon) should not rely on any particular semantics for this. This currently has:

~~~
if(PACKAGE_VERSION VERSION_LESS PACKAGE_FIND_VERSION)
  set(PACKAGE_VERSION_COMPATIBLE FALSE)
else()
  set(PACKAGE_VERSION_COMPATIBLE TRUE)
  if(PACKAGE_FIND_VERSION STREQUAL PACKAGE_VERSION)
    set(PACKAGE_VERSION_EXACT TRUE)
  endif()
endif()
~~~

Which should properly match with the current versioning scheme if we specify:

~~~
find_package(JANA 2.0.6 REQUIRED)
~~~

EXCEPT that we forgot to update the version number in the CMakeLists.txt, apparently for the last 2 releases:
https://github.com/JeffersonLab/JANA2/blob/e4d17a0b683198424ea28434d326f4a62e0c83a5/CMakeLists.txt#L2

(we'll fix that).
